context/amf_ue.go

```diff
@@ -302,12 +302,21 @@ func (ue *AmfUe) UnmarshalJSON(data []byte) error {
 	}
 
 	aux := auxCustom.CustomAmfUe
+	ran, ok := AMF_Self().AmfRanFindByAddr(aux.RanRemoteAddr)
+	if !ok {
+		logger.ContextLog.Warnln("Ran Connection is not Exist with GnbIp: ", aux.RanRemoteAddr)
+	}
 	for index, states := range aux.State {
-		ue.State[index].Set(fsm.StateType(states))
+		ue.State[index] = fsm.NewState(fsm.StateType(states))
 		if ue.RanUe[index] == nil {
 			ue.RanUe[index] = &RanUe{}
 		}
 		ue.RanUe[index].Log = logger.NgapLog.WithField(logger.FieldAmfUeNgapID, fmt.Sprintf("AMF_UE_NGAP_ID:%d", ue.RanUe[index].AmfUeNgapId))
+		ue.RanUe[index].RanUeNgapId = aux.RanUeNgapId
+		if ran != nil {
+			ran.RanUeList = append(ran.RanUeList, ue.RanUe[index])
+			ue.RanUe[index].Ran = ran
```

Comment:
I think we update the ran after we fetch the UE/RANUE. so we need to see how that affects the ranUelist append part.